src/Microsoft.PowerShell.ConsoleHost/host/msh/UpdatesNotification.cs

```diff
@@ -112,6 +112,13 @@ internal static async Task CheckForUpdates()
                 return;
             }
 
+            // Daily builds do not support update notifications
+            if (PSVersionInfo.PSCurrentVersion.PreReleaseLabel != null
```

Comment:
Thanks for catching this!
I think we should just update the `PSVersionInfo.GitCommitId.Length > 40` check to be `PSVersionInfo.GitCommitId.Length > 40 || PSVersionInfo.GitCommitId.IndexOf('daily') != -1`

Reply:
I can do this, but I'd argue that my implementation is not only easier to read but also faster because with your OR statement, both expressions have to be evaluated. In my implementation, I have an AND condition with `PSVersionInfo.PSCurrentVersion.PreReleaseLabel != null` being the first one being evaluated, which will always evaluate to false for release versions and does not require any function evaluation at all. What do you think?

Reply:
OK, but please 
1. move this check before the network check.
2. it's easier to read if you assign `PSVersionInfo.PSCurrentVersion.PreReleaseLabel` to a variable
3. compare with `"preview."` instead, so as to rule out other cases.

```c#
string preReleaseLabel = PSVersionInfo.PSCurrentVersion.PreReleaseLabel;
if (preReleaseLabel != null && !preReleaseLabel.StartsWith("preview.", StringComparison.OrdinalIgnoreCase))
{
    return;
}
```

Reply:
Ok. I agree with points 1 and 2.
Regarding point 3; if we compare against the string `preview` then wouldn't we also need to consider the case of the string `rc`?

Reply:
Ok. I agree with points 1 and 2.
Regarding point 3; if we compare against the string `preview` then wouldn't we also need to consider the case of the string `rc`?

Reply:
Good catch there. Then just go ahead compare with `daily` only :)

Reply:
Ok, done now